paymentsheet/src/test/java/com/stripe/android/customersheet/CustomerSheetTest.kt

```diff
@@ -0,0 +1,195 @@
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.intent.Intents.intended
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasExtra
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.R
+import com.stripe.android.model.CardBrand
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.Shadows.shadowOf
+import java.util.concurrent.CountDownLatch
+import kotlin.test.Test
+
+@OptIn(ExperimentalCustomerSheetApi::class)
+@RunWith(RobolectricTestRunner::class)
+class CustomerSheetTest {
+    @Before
+    fun setup() {
+        val appContext = ApplicationProvider.getApplicationContext<Application>()
+
+        val activityInfo = ActivityInfo().apply {
+            name = TestActivity::class.java.name
+            packageName = appContext.packageName
+            theme = R.style.StripePaymentSheetDefaultTheme
+        }
+
+        shadowOf(appContext.packageManager).addOrUpdateActivity(activityInfo)
+    }
+
+    @Test
+    fun `When presenting without configuring, should return error`() = runTestActivityTest { activity, complete ->
+        val customerSheet = CustomerSheet.create(
+            activity = activity,
+            customerAdapter = FakeCustomerAdapter(),
+            callback = { result ->
+                val failedResult = result.asFailed()
+
+                assertThat(failedResult.exception).isInstanceOf(IllegalStateException::class.java)
+                assertThat(failedResult.exception.message).isEqualTo(
+                    "Must call `configure` first before attempting to present `CustomerSheet`!"
+                )
+
+                complete()
+            },
+        )
+
+        customerSheet.present()
+    }
+
+    @Test
+    fun `When presenting, should launch 'CustomerSheetActivity'`() = runTestActivityTest { activity, complete ->
+        Intents.init()
+
+        val customerSheet = CustomerSheet.create(
+            activity = activity,
+            customerAdapter = FakeCustomerAdapter(),
+            callback = {},
+        )
+
+        val configuration = CustomerSheet.Configuration.builder(merchantDisplayName = "Merchant, Inc.")
+            .googlePayEnabled(googlePayConfiguration = true)
+            .preferredNetworks(preferredNetworks = listOf(CardBrand.CartesBancaires))
+            .build()
+
+        customerSheet.configure(
+            configuration = configuration,
+        )
+
+        customerSheet.present()
+
+        intended(hasComponent(CustomerSheetActivity::class.java.name))
+        intended(
+            hasExtra(
+                "args",
+                CustomerSheetContract.Args(
+                    configuration = configuration,
+                    statusBarColor = 0
+                )
+            )
+        )
+
+        Intents.release()
```

Comment:
Can we use [IntentsRule](https://developer.android.com/reference/androidx/test/espresso/intent/rule/IntentsRule) instead of manually doing these.

As is, if the tests fail, we won't release, which might cause other tests to fail and be harder to debug.